contracts/examples/multisig/tests/multisig_blackbox_test.rs

```diff
@@ -1,249 +1,589 @@
-    scenario_model::*,
+    scenario_model::{
+        Account, AddressValue, CheckAccount, CheckStateStep, ScCallStep, ScDeployStep, ScQueryStep,
+        SetStateStep, TxExpect,
+    },
     ContractInfo, ScenarioWorld,
 };
 use num_bigint::BigUint;
 
-const MULTISIG_PATH_EXPR: &str = "file:output/multisig.wasm";
+const ADDER_ADDRESS_EXPR: &str = "sc:adder";
+const ADDER_OWNER_ADDRESS_EXPR: &str = "address:adder-owner";
 const ADDER_PATH_EXPR: &str = "file:test-contracts/adder.wasm";
+const BOARD_MEMBER_ADDRESS_EXPR: &str = "address:board-member";
+const MULTISIG_ADDRESS_EXPR: &str = "sc:multisig";
+const MULTISIG_PATH_EXPR: &str = "file:output/multisig.wasm";
+const OWNER_ADDRESS_EXPR: &str = "address:owner";
+const PROPOSER_ADDRESS_EXPR: &str = "address:proposer";
+const QUORUM_SIZE: usize = 1;
+
+type MultisigContract = ContractInfo<multisig::Proxy<StaticApi>>;
+type AdderContract = ContractInfo<adder::Proxy<StaticApi>>;
 
 fn world() -> ScenarioWorld {
     let mut blockchain = ScenarioWorld::new();
     blockchain.set_current_dir_from_workspace("contracts/examples/multisig");
 
-    blockchain.register_contract(ADDER_PATH_EXPR, adder::ContractBuilder);
     blockchain.register_contract(MULTISIG_PATH_EXPR, multisig::ContractBuilder);
+    blockchain.register_contract(ADDER_PATH_EXPR, adder::ContractBuilder);
     blockchain
 }
 
-#[test]
-fn basic_setup_test() {
-    let mut test = MultisigTestState::setup();
-    test.multisig_deploy();
-
-    // TODO: transform into a framework test and replace here with sc_call_get_result
-    let mut board_members = MultiValueVec::<Address>::new();
-    test.world.sc_call_use_result(
-        ScCallStep::new()
-            .from(&test.alice)
-            .call(test.multisig.get_all_board_members()),
-        |tr| {
-            board_members = tr.result.unwrap();
-        },
-    );
-
-    let expected_board_members: Vec<_> = [
-        test.alice.to_address(),
-        test.bob.to_address(),
-        test.carol.to_address(),
-    ]
-    .into();
-
-    assert_eq!(board_members.into_vec(), expected_board_members);
-}
-
-#[test]
-fn multisig_adder_test() {
-    let mut test = MultisigTestState::setup();
-    test.adder_deploy().multisig_deploy();
-
-    let caller = &test.alice.to_address();
-    let signers = [
-        &test.alice.to_address(),
-        &test.bob.to_address(),
-        &test.carol.to_address(),
-    ];
-    let deployed_sc_address = test.multisig_deploy_adder(caller, signers.as_slice());
-    assert_eq!(deployed_sc_address, test.adder_multisig.to_address());
-
-    let first_value: BigUint = 42u64.into();
-    let second_value: BigUint = 43u64.into();
-    let expected_sum = first_value.clone() + second_value.clone();
-
-    test.multisig_call_adder_add(first_value, caller, &signers);
-    test.multisig_call_adder_add(second_value, caller, &signers);
-    test.adder_expect_get_sum(expected_sum, caller);
+enum Action {
```

Comment:
The additional enum helpful in the whitebox tests, but in the blackbox tests it is quite straightforward to use Action<StaticApi> directly from the contract.